test/integration/src/test/java/io/pravega/test/integration/MetricsTest.java

```diff
@@ -0,0 +1,293 @@
+        }
+        if (this.server != null) {
+            this.server.close();
+            this.server = null;
+        }
+        if (this.zkTestServer != null) {
+            this.zkTestServer.close();
+            this.zkTestServer = null;
+        }
+
+    }
+
+    @Test
+    public void metricsTimeBasedCacheEvictionTest() throws InterruptedException, ExecutionException {
+
+        try (StreamManager streamManager = new StreamManagerImpl(controller)) {
+
+            Boolean createScopeStatus = streamManager.createScope(scope);
+            log.info("Create scope status {}", createScopeStatus);
+
+            Boolean createStreamStatus = streamManager.createStream(scope, STREAM_NAME, config);
+            log.info("Create stream status {}", createStreamStatus);
+
+        }
+
+        try (ConnectionFactory connectionFactory = new ConnectionFactoryImpl(ClientConfig.builder().build());
+             ClientFactory clientFactory = new ClientFactoryImpl(scope, controller, connectionFactory);
+             ReaderGroupManager readerGroupManager = new ReaderGroupManagerImpl(scope, controller, clientFactory, connectionFactory)) {
+
+            EventStreamWriter<String> writer1 = clientFactory.createEventWriter(STREAM_NAME,
+                    new JavaSerializer<String>(),
+                    EventWriterConfig.builder().build());
+
+            String event = "12345";
+            long bytesWritten = TOTAL_NUM_EVENTS * event.length() * 4;
+
+            for (int i = 0; i < TOTAL_NUM_EVENTS; i++) {
+                try {
+                    log.info("Writing event {}", event);
+                    writer1.writeEvent("", event);
+                    writer1.flush();
+                } catch (Throwable e) {
+                    log.warn("Test exception writing events: {}", e);
+                    break;
+                }
+            }
+
+            String readerGroupName1 = readerGroupName + "1";
+            log.info("Creating Reader group : {}", readerGroupName1);
+
+            readerGroupManager.createReaderGroup(readerGroupName1, ReaderGroupConfig.builder().stream(Stream.of(scope, STREAM_NAME)).build());
+
+            EventStreamReader<String> reader1 = clientFactory.createReader(readerName,
+                    readerGroupName1,
+                    new JavaSerializer<String>(),
+                    ReaderConfig.builder().build());
+
+            for (int j = 0; j < TOTAL_NUM_EVENTS; j++) {
+
+                try {
+                    String eventRead1 = reader1.readNextEvent(SECONDS.toMillis(2)).getEvent();
+                    log.info("Reading event {}", eventRead1);
+                } catch (ReinitializationRequiredException e) {
+                    log.warn("Test Exception while reading from the stream", e);
+                }
+
+            }
+
+            long initialCount = MetricsProvider.METRIC_REGISTRY.getCounters().get("pravega.segmentstore.segment_read_bytes." + scope + "." + STREAM_NAME + ".0.Counter").getCount();
```

Comment:
create a method to generate this name as you are going to use it at several places.

Reply:
done.